selfdrive/ui/qt/widgets/controls.h

```diff
@@ -188,89 +189,61 @@ class ParamControl : public ToggleControl {
     } else if (!icon_pixmap.isNull()) {
       icon_label->setPixmap(icon_pixmap);
     }
-  };
+  }
 
   std::string key;
   Params params;
   QPixmap active_icon_pixmap;
   bool confirm = false;
   bool store_confirm = false;
 };
- 
+
 class ButtonParamControl : public AbstractControl {
   Q_OBJECT
 public:
   ButtonParamControl(const QString &param, const QString &title, const QString &desc, const QString &icon,
-                     std::vector<QString> button_texts, std::vector<int> button_widths) : AbstractControl(title, desc, icon) {
-    select_style = (R"(
-        padding: 0;
+                     const std::vector<QString> &button_texts) : AbstractControl(title, desc, icon) {
+    const QString style = R"(
+      QPushButton {
         border-radius: 50px;
-        font-size: 45px;
+        font-size: 40px;
         font-weight: 500;
-        color: #E4E4E4;
-        background-color: #33Ab4C;
-      )");
-    unselect_style = (R"(
-        padding: 0;
-        border-radius: 50px;
-        font-size: 45px;
-        font-weight: 350;
+        height:100px;
+        padding: 0 25 0 25;
         color: #E4E4E4;
         background-color: #393939;
-      )");
+      }
+      QPushButton:pressed {
+        background-color: #4a4a4a;
+      }
+      QPushButton:checked {
+        background-color: #33Ab4C;
+      }
+    )";
     key = param.toStdString();
+    int value = atoi(params.get(key).c_str());
+
+    QButtonGroup *button_gorup = new QButtonGroup(this);
+    button_gorup->setExclusive(true);
     for (int i = 0; i < button_texts.size(); i++) {
-      QPushButton *button = new QPushButton();
-      button->setText(button_texts[i]);
+      QPushButton *button = new QPushButton(button_texts[i], this);
+      button->setCheckable(true);
+      button->setChecked(i == value);
+      button->setStyleSheet(style);
       hlayout->addWidget(button);
-      button->setFixedSize(button_widths[i], 100);
-      button->setStyleSheet(unselect_style);
-      buttons.push_back(button);
-      QObject::connect(button, &QPushButton::clicked, [=]() {
-        params.put(key, (QString::number(i)).toStdString());
-        refresh();
-      });
-    }
-    refresh();
-  }
-
-  void read_param() {
-    auto value = params.get(key);
-    if (!value.empty()) {
-      param_value = std::stoi(value);
+      button_gorup->addButton(button, i);
     }
-  };
-  void set_param(int new_value) {
-    QString values = QString::number(new_value);
-    params.put(key, values.toStdString());
-    refresh();
-  };
 
-  void refresh() {
-    read_param();
-    for (int i = 0; i < buttons.size(); i++) {
-      buttons[i]->setStyleSheet(unselect_style);
-      if (param_value == i) {
-        buttons[i]->setStyleSheet(select_style);
-      }
-    }
-  };
-  void showEvent(QShowEvent *event) override {
-    refresh();
-  };
+    QObject::connect(button_gorup, QOverload<int>::of(&QButtonGroup::buttonClicked), [=](int id) {
```

Comment:
should be `buttonToggled` so we don't keep writing if the user taps the current button?